subsys/net/l2/wifi/wifi_shell.c

```diff
@@ -1127,6 +1127,98 @@ static int cmd_wifi_ps_wakeup_mode(const struct shell *sh, size_t argc, char *ar
 	return 0;
 }
 
+static int cmd_wifi_promisc_mode(const struct shell *sh, size_t argc, char *argv[])
+{
+	struct net_if *iface = net_if_get_default();
+	struct wifi_promisc_setup promisc_mode = {0};
+	int ret;
+	long mode;
+
+	if (argc == 1) {
+		promisc_mode.oper = WIFI_MGMT_GET;
+	} else if (argc == 2) {
+		promisc_mode.oper = WIFI_MGMT_SET;
+		if (!parse_number(sh, &mode, argv[1], WIFI_PROMISC_DISABLE,
+				WIFI_PROMISC_STA_AP_MODE)) {
+			return -EINVAL;
+		}
+		promisc_mode.mode =  mode;
+	} else {
+		return -EINVAL;
+	}
+
+	promisc_mode.op = WIFI_PROMISC_MODE;
+	ret = net_mgmt(NET_REQUEST_WIFI_PROMISC_SETUP, iface,
+			&promisc_mode, sizeof(promisc_mode));
+	if (ret) {
+		shell_fprintf(sh, SHELL_ERROR, "Unable to perform %s promisc mode operation: %d\n",
```

Comment:
I think that for the help strings/error messages we should not use the abbreviated form, please change to `promiscuous` here and in other error messages below.

Reply:
will make the relevant change